frontend/pages/event-recaps/[year]/[event_recap_id].js

```diff
@@ -9,79 +9,52 @@ import Typography from "@material-ui/core/Typography";
 import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
 import AwesomeSlider from "react-awesome-slider";
 import "react-awesome-slider/dist/styles.css";
-import axios from "axios";
-import humps from "humps";
 import Image from "next/image";
+import { loadPastEvents } from "../../../lib/api";
 
-const EventRecapPage = (props) => {
+const EventRecapPage = ({ selectedEvent }) => {
   const [expanded, setExpanded] = useState(false);
-  const [event, setEvent] = useState({});
+  const [event, setEvent] = useState(selectedEvent);
   const [loading, setLoading] = useState(true);
   const [hasResources, setHasResources] = useState(false);
   const [hasPhotos, setHasPhotos] = useState(false);
   const [hasFBLink, setHasFBLink] = useState(false);
   const [imageGalleryFilenames, setImageGalleryFilenames] = useState([]);
   const [hasEmbeddedVideo, setHasEmbeddedVideo] = useState(false);
   const [enableGalleryArrows, setEnableGalleryArrows] = useState(true);
-  const [eventNumber, setEventNumber] = useState("0");
 
-  useEffect(() => window.scrollTo(0, 0), []);
+  const fetchPastEvent = (event) => {
+    setLoading(false);
+    setPhotos(event);
 
-  useEffect(() => {
-    setLoading(true);
-
-    let url = window.location.href.split("/");
-    setEventNumber(url[url.length - 1]);
-    let currEventNumber = url[url.length - 1];
-
-    const fetchPastEvents = async () => {
-      const res = await axios.get(
-        "https://wit-database.herokuapp.com/past-events"
-      );
-      const allEvents = humps
-        .camelizeKeys(res.data)
-        .filter((event) => event.year.toString() === url[url.length - 2]);
-
-      if (allEvents.length <= 0 || currEventNumber > allEvents.length) {
-        props.history.push("/404");
-        return;
-      }
-
-      // load the page content for the current event
-      const currEvent = allEvents.filter((event) => {
-        return event.eventNumber.toString() === currEventNumber;
-      })[0];
-      setEvent(currEvent);
-
-      // hide the loading sign
-      setLoading(false);
-
-      setPhotos(currEvent);
+    if (event.fields.resourcesFolderId) {
```

Comment:
Redirect to 404 page if url is invalid e.g. 2021/400. Currently, the browser throws “application error” when this behaviour occurs